articles/flask-database-integration-with-sqlalchemy/index.md

```diff
@@ -0,0 +1,329 @@
+
+```python
+ from app import db
+```
+
+Next let’s create the User model to handle the registration form data for the username, email, and password.
+
+```Python 
+ from app import db
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key =True)
+    username = db.Column(db.String(100), nullable = False)
+    user_email = db.Column(db.String(100), nullable= False)
+    user_password = db.Column(db.String(150), nullable = False)
+```
+In the model declaration, you can observe the arguments passed to the String data type, they simply are the length of data I expect the user to not surpass. 
+
+Next, in our __init__.py file, beneath the database object initialization, we import the models file. The reason for this is that when we create our database, Flask will not be able to access the database models in our models.py file, therefore we have to draw it back into the app initialization file for the database object to be aware of its existence.
+
+```python
+from app import models
+```
+At this point, your __init__.py file should look like this:
+
+```python
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+
+app = Flask(__name__)
+
+app.config["SECRET_KEY"] = '571ebf8e13ca209536c29be68d435c00'
+app.config["SQLALCHEMY_DATABASE_URI"] = 'sqlite:///registration.db'
+db = SQLAlchemy(app)
+
+from app import views
+from app import models
+```
+
+And your models.py file should look exactly like this;
+
+```python
+from app import db
+ 
+ 
+ 
+class User(db.Model):
+   id = db.Column(db.Integer, primary_key =True)
+   username = db.Column(db.String(100), nullable = False)
+   user_email = db.Column(db.String(100), nullable= False)
+   user_password = db.Column(db.String(150), nullable = False)
+
+```
+
+We have the data models created, let’s now create this database for usage, to do this, we need to open our terminal to our working directory and follow this process:
+
+Initialize the Python interpreter
+```bash
+$ python
+```
+Next, import the database variable from the __init__.py file where we initialized the SQLAlchemy instance of the object, then create the database.
+
+```bash
+$ >> from app import db
+$>> db.create_all()
+```
+Remember that the database variable is the variable name we used in our __init__.py, if you intend to use a different variable, you have to call that variable name itself and not the database variable created here.
+
+To confirm our database is created, check your app folder, you should see a new file title registration.db, which is the db name we specified in our __init__.py file.
+
+Next, we can import the User table in our models file, via this command
+
+```bash
+$>> from app.models import User
+```
+ To ensure the functionality of the database is intact, let’s create a dummy user from the python interpreter
+
+```bash
+>>> user1 = User(username = "Demo User", user_email = "demouser@gmail.com", user_password = "demouserpassword")
```

Comment:
For certain bash commands you have used `$>>` and for a few, you have used `>>>`.

Make sure it's consistent throughout the article. Make necessary changes everywhere

Reply:
I would suggest to remove `$` from all the bash commands.

Reply:
@corpsgeek